bot/src/main/java/com/community/tools/service/github/PullRequestValidationService.java

```diff
@@ -0,0 +1,161 @@
+                + "        line: ${line_number_integer},"
+                + "        comment: ${comment}"
+                + "      }"
+                + "    ]"
+                + "  ]"
+                + "} return only strictly json response.";
+
+        requests.add(QueuedValidationProcessDto.builder()
+                .prompt(prompt)
+                .pullRequest(pullRequest)
+                .commitDetail(commitDetail)
+                .fileCodeLines(fileList.stream()
+                        .collect(
+                                Collectors.toMap(
+                                        CommitFileDto::getFilename,
+                                    commitFileDto -> commitFileDto.getCode().split("\n").length
+                                )
+
+                        )
+                )
+                .build());
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Method that processes each request in the request queue,
+   * sends it to OpenAI Api, so it could validate it
+   * and leave a review in the PR based on the AI response.
+   * Method is scheduled to be called each 21 seconds to prevent quota violation.
+   *
+   * @throws RuntimeException If github api doesn't find requested objects
+   */
+  @Scheduled(cron = "*/21 * * * * *")
+  private void processValidation() {
+    if (!requests.isEmpty()) {
+      QueuedValidationProcessDto process = requests.poll();
+      String prompt = process.getPrompt();
+      GHPullRequest pullRequest = process.getPullRequest();
+      GHPullRequestCommitDetail commitDetail = process.getCommitDetail();
+      GHPullRequestReviewBuilder review = pullRequest.createReview();
+      try {
+        String response = openAiService.processPrompt(prompt);
+        OpenAiValidationResponseDto openAiResponse = objectMapper.readValue(response,
+                OpenAiValidationResponseDto.class);
+        try {
+          openAiResponse.getFiles().forEach(file -> {
+            int codeLines = process.getFileCodeLines().get(file.getFilename());
+            file.getComments().forEach(comment -> {
+              if (codeLines > comment.getLine()) {
+                review.commitId(commitDetail.getSha());
+                review.comment(comment.getComment(), file.getFilename(), comment.getLine());
+              }
+            });
+          });
+          if (openAiResponse.getRating() > 7) {
+            review.event(GHPullRequestReviewEvent.COMMENT);
+          } else {
+            review.event(GHPullRequestReviewEvent.REQUEST_CHANGES);
+          }
+          review.body("OpenAI previous review of commit "
+                  + commitDetail.getSha() + ". The rating is "
+                  + openAiResponse.getRating() + "/10.");
+
+          review.create();
+        } catch (IOException e) {
+          throw new RuntimeException(e);
+        }
+      } catch (JsonProcessingException exception) {
+        requests.add(process);
```

Comment:
We should add explanation, why we are doing this

Reply:
Added explanation to the Javadoc